examples/sample_runtime/src/main.rs

```diff
@@ -176,37 +182,32 @@ impl Runtime for SampleRuntime {
             }
 
             // Unknown transaction.
-            (interface, method) => Err((
-                SampleRuntimeError::IncorrectCallInfo,
-                format!(
-                    "Incorrect information to call transaction. {}#{}",
-                    interface, method
-                ),
-            )
-                .into()),
+            (interface, method) => {
+                let err = (
+                    SampleRuntimeError::IncorrectCallInfo,
+                    format!(
+                        "Incorrect information to call transaction. {}#{}",
+                        interface, method
+                    ),
+                );
+                Err(err.into())
+            }
         }
     }
 
-    fn artifact_protobuf_spec(&self, id: &ArtifactId) -> Option<ArtifactProtobufSpec> {
-        self.deployed_artifacts
-            .get(id)
-            .map(|_| ArtifactProtobufSpec::default())
-    }
-
     fn state_hashes(&self, _snapshot: &dyn Snapshot) -> StateHashAggregator {
         StateHashAggregator::default()
     }
 
-    fn before_commit(&self, _dispatcher: &DispatcherRef, _fork: &mut Fork) {}
-
-    fn after_commit(
+    fn before_commit(
         &self,
-        _dispatcher: &DispatcherSender,
-        _snapshot: &dyn Snapshot,
-        _service_keypair: &(PublicKey, SecretKey),
-        _tx_sender: &ApiSender,
-    ) {
+        _context: ExecutionContext,
+        _id: InstanceId,
+    ) -> Result<(), ExecutionError> {
+        Ok(())
     }
+
+    fn after_commit(&mut self, _snapshot: &dyn Snapshot, _mailbox: &mut Mailbox) {}
 }
 
 impl From<SampleRuntime> for (u32, Box<dyn Runtime>) {
```

Comment:
IMO, this conversion looks too generic / non-obvious. It could make sense to introduce a special-purpose trait like

```rust
trait RuntimeWithId: Runtime {
    const ID: u32;
}
```

Reply:
Same for `impl From<ServiceName> for InstanceCollection`.